force-app/main/default/classes/FieldBucketSelector.cls

```diff
@@ -0,0 +1,39 @@
+/*
+ *
+ *  * Copyright (c) 2022, salesforce.com, inc.
+ *  * All rights reserved.
+ *  * SPDX-License-Identifier: BSD-3-Clause
+ *  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ *
+ */
+
+public with sharing class FieldBucketSelector {
+    public List<Bucket__mdt> getBuckets(
+        List<String> bucketNames,
+        Schema.SObjectType objectType,
+        Schema.SObjectField bucketedField
+    ) {
+        // Note: there doesn't seem to be a way to get the subquery with .getAll() so I think we're stuck
+        // with SOQL for this layer.
+
+        String sObjectAPIName = objectType.getDescribe().getName();
+        //String bucketedFieldAPIName = bucketedField.getDescribe().getName();
```

Comment:
This line was throwing a Null-Pointer Exception in 2gp orgs, which leads me to think that there's some kind of namespace issue with getting the SObjectField - however, I would have thought that since all this code is running within package context the namespace wouldn't be required. It's currently not needed because the part of the WHERE clause where it was being used was also causing a server error (see below) so neither are currently used, but I think I need to spin up a 2gp org and figure out what's going on. 

That said, if anyone has suggestions I'm welcome to entertain them!